scale-encode/src/error/mod.rs

```diff
@@ -91,93 +112,68 @@ impl Display for Error {
 }
 
 /// The underlying nature of the error.
-#[derive(Debug)]
+#[derive(Debug, derive_more::From, derive_more::Display)]
 pub enum ErrorKind {
     /// Cannot find a given type.
+    #[display(fmt = "Cannot find type with ID {_0}")]
     TypeNotFound(u32),
     /// Cannot encode the actual type given into the target type ID.
+    #[display(fmt = "Cannot encode {actual:?} into type with ID {expected}")]
     WrongShape {
         /// The actual kind we have to encode
         actual: Kind,
         /// ID of the expected type.
         expected: u32,
     },
     /// The types line up, but the expected length of the target type is different from the length of the input value.
+    #[display(
+        fmt = "Cannot encode to type; expected length {expected_len} but got length {actual_len}"
+    )]
     WrongLength {
         /// Length we have
         actual_len: usize,
         /// Length expected for type.
         expected_len: usize,
     },
     /// We cannot encode the number given into the target type; it's out of range.
+    #[display(fmt = "Number {value} is out of range for target type {expected}")]
```

Comment:
minor nit: we could keep consistency with [commit](https://github.com/paritytech/scale-decode/pull/31/commits/e70634dba33d422d22e66e256dec71a8755eddae), although I like it how it is right now

Reply:
Actually, I undid the other ones because I realised they weren't really necessary!